Sources/KlaviyoSwift/KlaviyoModels.swift

```diff
@@ -8,29 +8,30 @@
 import AnyCodable
 import Foundation
 
+public protocol MetricNameProtocol {
+    var value: String { get }
+}
+
 public struct Event: Equatable {
-    public enum EventName: Equatable {
-        case OpenedPush
-        case ViewedProduct
-        case SearchedProducts
-        case StartedCheckout
-        case PlacedOrder
-        case OrderedProduct
-        case CancelledOrder
-        case PaidForOrder
-        case SubscribedToBackInStock
-        case SubscribedToComingSoon
-        case SubscribedToList
-        case SuccessfulPayment
-        case FailedPayment
-        case CustomEvent(String)
+    public struct Legacy: Equatable {
+        public enum MetricName: MetricNameProtocol {
+            case OpenedPush
+            case ViewedProduct
+            case StartedCheckout
+            case OpenedApp
+            case AddedToCart
+            case CustomEvent(String)
+        }
     }
 
-    public struct Metric: Equatable {
-        public let name: EventName
-
-        public init(name: EventName) {
-            self.name = name
+    public struct V1: Equatable {
```

Comment:
Sorry late to this PR - do we need the V1 layer?

Reply:
for RN no...

Reply:
No I meant in general - do we need this since I feel like it gives the impression that these properties are things that change frequently. This is a nit comment nothing blocking.

Reply:
Oh agreed. We needed a way to get people to update their code. This seemed like the easiest way to make that happen. If we leave it as is someone might end up with duplicate event names. If they want to preserve the old string name they can use v1.

Reply:
That makes sense.